students/kuhnbt/lesson9/oo_mailroom.py

```diff
@@ -0,0 +1,170 @@
+                    self.donor_collection.add_donor(current_donor)
+
+                current_donor.add_donation(donation)
+
+                print(current_donor.get_thank_you())
+                break
+        return
+
+    def create_report(self):
+        """Print summary report of donor info"""
+        max_donor_width = max([len(donor.name) for donor in
+                               self.donor_collection.donors])
+        print(self.get_report_header(max_donor_width))
+        self.donor_collection.sort_donors()
+        for donor in self.donor_collection.donors:
+            print(donor.get_report_row(max_donor_width))
+
+    def get_report_header(self, max_donor_width):
+        """Return string representing the header of the report"""
+        return '{:{}}|{:12}|{:10}|{:8}'.format('Donor Name',
+                                               max_donor_width, 'Total Given',
+                                               'Num Gifts', 'Average Gift')
+
+    def send_to_everyone(self, directory=os.getcwd()):
+        """Write thank you notes to specified directory for each donor in
+           donor_info"""
+        user_dir = input('Change output directory (y/n)?')
+        if user_dir.lower() == 'y':
+            while True:
+                directory = input('Please enter desired directory:')
+                try:
+                    os.chdir(directory)
+                    break
+                except FileNotFoundError:
+                    print('Please enter a valid directory')
+                except TypeError:
+                    print('Please enter a valid directory')
+
+        today = datetime.datetime.today().strftime('%Y-%m-%d')
+        for donor in self.donor_collection.donors:
+            filename = donor.name + '_' + today + '.txt'
+            with open(filename, 'w') as f:
+                print('Writing to ' + filename + '...')
+                f.write(donor.get_thank_you())
+            print('Finished!')
+
+    def quit_program(self):
+        """Exit program"""
+        sys.exit()
+
+
+class DonorCollection:
+    def __init__(self, donors=[]):
+        self.donors = []
+        for donor in donors:
+            self.add_donor(donor)
+
+    def add_donor(self, donor):
+        if not isinstance(donor, Donor):
+            raise TypeError('Items in DonorCollection must be of type '
+                            'Donor')
+        self.donors.append(donor)
+
+    def sort_donors(self):
+        self.donors.sort()
+
+
+class Donor:
+
+    def __init__(self, name, donations=[]):
+        self.name = name
+        self.donations = []
+        self.average_donation = 0
+        self.get_average_donation()
+
+    def add_donation(self, donation):
+        if type(donation) == list:
+            for l in donation:
+                self.donations.append(float(l))
+        else:
+            self.donations.append(float(donation))
+        self.get_average_donation()
+
+    def get_average_donation(self):
```

Comment:
this should be called `set_average_donation` since you're not actually getting a value out of this function.